### PR TITLE
Add typing for builtin

### PR DESCRIPTION
**Pull request by Jc2k**

### CODE REVIEW

File: music_assistant/server/providers/builtin/__init__.py [hunk 1]
```diff
@@ -507,59 +524,76 @@ async def get_stream_details(self, item_id: str) -> StreamDetails:
...
-                return result
-        if builtin_playlist_id == RECENTLY_PLAYED:
-            tracks = await self.mass.music.recently_played(100, [MediaType.TRACK])
             for idx, track in enumerate(tracks, 1):
                 track.position = idx
                 result.append(track)
-            return result
         return result
 
+    async def _get_builtin_playlist_recently_played(self) -> list[Track]:
```
Comment (Jc2k) [inline music_assistant/server/providers/builtin/__init__.py:575]: None of this code actually changed, but mypy had a real hard time with reusing "result" in different branches of the if statement and getting slightly different signatures (i guess another case of lists == invariant == bad?).

File: music_assistant/server/providers/builtin/__init__.py [hunk 2]
```diff
@@ -507,59 +524,76 @@ async def get_stream_details(self, item_id: str) -> StreamDetails:
...
+            result.append(item)
+        return result
+
+    async def _get_builtin_playlist_random_album(self) -> list[Track]:
+        result: list[Track] = []
+        for random_album in await self.mass.music.albums.library_items(
+            limit=1, order_by="random_fast"
+        ):
+            # use the function specified in the queue controller as that
+            # already handles unwrapping an album by user preference
```
Comment (marcelveldt) [inline music_assistant/server/providers/builtin/__init__.py:551]:
```suggestion
```
outdated comment as that function moved into the albums controller

File: music_assistant/server/providers/builtin/__init__.py [hunk 3]
```diff
@@ -507,59 +524,76 @@ async def get_stream_details(self, item_id: str) -> StreamDetails:
...
+                result.append(track)
+        return result
+
+    async def _get_builtin_playlist_random_artist(self) -> list[Track]:
+        result: list[Track] = []
+        for random_artist in await self.mass.music.artists.library_items(
+            limit=1, order_by="random_fast"
+        ):
+            # use the function specified in the queue controller as that
+            # already handles unwrapping an artist by user preference
```
Comment (marcelveldt) [inline music_assistant/server/providers/builtin/__init__.py:566]:
```suggestion
```

File: music_assistant/server/providers/builtin/__init__.py (hunk 1 above)
Comment (marcelveldt) [reply music_assistant/server/providers/builtin/__init__.py:575]: It looks much better now - nice cleanup/restructure

File: music_assistant/server/providers/builtin/__init__.py [hunk 4]
```diff
@@ -507,59 +524,76 @@ async def get_stream_details(self, item_id: str) -> StreamDetails:
...
+            return await {
+                ALL_FAVORITE_TRACKS: self._get_builtin_playlist_random_favorite_tracks,
+                RANDOM_TRACKS: self._get_builtin_playlist_random_tracks,
+                RANDOM_ALBUM: self._get_builtin_playlist_random_album,
+                RANDOM_ARTIST: self._get_builtin_playlist_random_artist,
+                RECENTLY_PLAYED: self._get_builtin_playlist_recently_played,
+            }[builtin_playlist_id]()
+        except KeyError:
+            raise MediaNotFoundError(f"No built in playlist: {builtin_playlist_id}")
+
```
Comment (marcelveldt) [inline music_assistant/server/providers/builtin/__init__.py:596]: fancy

**Review by marcelveldt — APPROVED** [review]
Nice!

**Pull request merged**